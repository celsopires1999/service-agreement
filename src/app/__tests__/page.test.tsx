import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import Home from "../page"

describe("Page", () => {
    it("renders the main heading", () => {
        render(<Home />)

        const heading = screen.getByRole("heading", {
            level: 1,
            name: /service agreement validation/i,
        })

        expect(heading).toBeInTheDocument()
    })

    it("renders the secondary heading and paragraph", () => {
        render(<Home />)

        const secondaryHeading = screen.getByRole("heading", {
            level: 2,
            name: /your service provider at your disposal/i, // 'name' handles text broken by <br>
        })
        // When text is split by <br />, it's better to find parts of it
        const p_line1 = screen.getByText(/daily availability/i)
        const p_line2 = screen.getByText(/8am to 5pm/i)

        expect(secondaryHeading).toBeInTheDocument()
        expect(p_line1).toBeInTheDocument()
        expect(p_line2).toBeInTheDocument()
    })

    it("renders the start link pointing to the agreements page", () => {
        render(<Home />)
        const link = screen.getByRole("link", { name: /start/i })
        expect(link).toBeInTheDocument()
        expect(link).toHaveAttribute("href", "/agreements")
    })

    it("renders homepage unchanged", () => {
        const { container } = render(<Home />)
        expect(container).toMatchSnapshot()
    })
})
