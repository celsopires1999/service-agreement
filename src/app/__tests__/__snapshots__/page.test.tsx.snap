// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Page renders homepage unchanged 1`] = `
<div>
  <div
    class="bg-black bg-home-img bg-cover bg-center"
  >
    <main
      class="mx-auto flex h-dvh max-w-5xl flex-col items-center justify-center text-center"
    >
      <div
        class="mx-auto flex w-4/5 flex-col gap-6 rounded-xl bg-black/90 p-12 text-white sm:max-w-96 sm:text-2xl"
      >
        <h1>
          Service Agreement
          <br />
          Validation
        </h1>
        <h2>
          Your Service Provider
          <br />
          At your disposal
        </h2>
        <p>
          Daily Availability
          <br />
          8am to 5pm
        </p>
        <a
          class="hover:underline"
          href="/agreements"
        >
          Start
        </a>
      </div>
    </main>
  </div>
</div>
`;
